### PR TITLE
chore: check all errors in tests

### PR DESCRIPTION
**Pull request by faddat**
Closes: #5060 

## What is the purpose of the change

This PR is designed to enhance the rigor of tests in Osmosis by **checking errors whenever possible**. 

It also extends the changes found in:

* #5064 

Note that the t.Helper() thing can be a bit all or nothing, so this PR contains those changes as well.  If you'd prefer to close that one, no worries :). 

By applying thelper to the whole repository.  This is a part of an initiative to allow Osmosis to use golangci-lint on its tests, by setting 

```yaml
run:
  tests: true
  timeout: 10m
```

and then progressively adding additional tools like thelper.  Note to vscode users: now that we're using t.Helper() at the beginning of every test helper, you'll get these really cool inline stack traces if there's an issue with a test. 

Since we want golangci-lint to pass, this PR does not yet make the change to the configuration mentioned above, but it is what was used while writing the PR. 

## Brief Changelog

* check as many errors as possible
* minor linting on tests, only as a part of the error work

## Testing and Verifying

This change **is** tests, and I suppose that the best way to test it is to maybe break some tests and check out the cool stack trace?  I am actually not certain, but would love to hear ideas. 

## Reviewing

I'd recommend reviewing 

* https://github.com/osmosis-labs/osmosis/pull/5064

before this one, it will reduce the size of this.  To make sure that this merges smoothly, I cherry picked that into this. 

There are about 200-300 miscellaneous linter changes that Id need to make for us to be able to enable the linter on the tests.  I'm going to open one or more additional smaller PR's for that since I know this one is already pretty big. 

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/concentrated-liquidity/position_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/position_test.go:635]:
```suggestion
			s.PrepareConcentratedPool()
```

**Review by czarcas7ic — APPROVED** [review]
Approving, but going to wait for one more internal ack before merging. Thanks!

**Review by p0mvn — APPROVED** [review]
LGTM, thanks for this 🧹

**Comment by p0mvn** [comment]
Ready to merge once few nits are addressed and PR is refreshed against `main`

**Comment by faddat** [comment]
Cool, refreshing now :)

File: x/concentrated-liquidity/export_test.go
Comment (faddat) [reply x/concentrated-liquidity/export_test.go:220]: sure :)

File: .golangci.yml
Comment (faddat) [reply .golangci.yml:3]: actually no.

**Pull request merged**
